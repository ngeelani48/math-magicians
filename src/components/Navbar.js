import { Link, Outlet } from 'react-router-dom';

const Navbar = () => (
  <nav className="nav">
    <h1 className="main-title">Math Magicians</h1>
    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <hr />
      <li>
        <Link to="/calculator">Calculator</Link>
      </li>
      <hr />
      <li>
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
    <Outlet />
  </nav>
);

export default Navbar;
