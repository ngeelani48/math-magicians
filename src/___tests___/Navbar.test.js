import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from "../components/Navbar";

it ('check if the Navbar section renders properly', ()=>{
    const tree = render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(tree).toMatchSnapshot();
});
