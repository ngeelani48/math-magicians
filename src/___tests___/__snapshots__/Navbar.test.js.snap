// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if the Home section renders properly 1`] = `
<div
  className="home-container"
>
  <h2
    className="home-title"
  >
    Welcome to my Math Magicians website!
  </h2>
  <div
    className="homeParas"
  >
    <p
      className="homeP"
    >
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum!
    </p>
    <p
      className="homeP"
    >
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Facilis minima explicabo ex laboriosam vitae quas repudiandae culpa quam labore omnis est, laudantium enim in nulla eius inventore neque odit eum!
    </p>
  </div>
</div>
`;

exports[`check if the Navbar section renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <h1
          class="main-title"
        >
          Math Magicians
        </h1>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <hr />
          <li>
            <a
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <hr />
          <li>
            <a
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <h1
        class="main-title"
      >
        Math Magicians
      </h1>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <hr />
        <li>
          <a
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <hr />
        <li>
          <a
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
